src/components/Comments.jsx

```diff
@@ -1,15 +1,36 @@
 import moment from "moment";
-import React, { useEffect, useState } from "react";
+import React, { useContext, useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
-import { getComments } from "../utils/api";
+import { deleteComment, getComments } from "../utils/api";
 import CommentVotes from "./CommentVotes";
 import PostComment from "./PostComment";
+import { UserContext } from "../contexts/users";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { solid } from "@fortawesome/fontawesome-svg-core/import.macro";
 
 export default function Comments() {
   const [comments, setComments] = useState([]);
   const [isLoading, setIsLoading] = useState(false);
   const { review_id } = useParams();
   const [newCommentCount, setNewCommentCount] = useState(0);
+  const { user } = useContext(UserContext);
+  const [error, setError] = useState("");
+
+  const handleDelete = (commentToDelete) => {
+    setComments((currComments) => {
+      return currComments.filter((comment) => {
+        return comment.comment_id !== commentToDelete.comment_id;
+      });
+    });
+    deleteComment(commentToDelete.comment_id).catch(() => {
+      setError("Something went wrong, please try again.");
+      setComments((currComments) => {
+        const newComments = [...currComments];
+        newComments.unshift(commentToDelete);

```

Comment:
This is ok but will disrupt the order of comments, could you instead save a copy of the original comments to preserve the order?